src/transformers/modeling_utils.py

```diff
@@ -2575,8 +2587,10 @@ def save_pretrained(
             if safe_serialization:
                 # At some point we will need to deal better with save_function (used for TPU and other distributed
                 # joyfulness), but for now this enough.
-                safe_save_file(shard, os.path.join(save_directory, shard_file), metadata={"format": "pt"})
+                metadata.update({"format": "pt"})
+                safe_save_file(shard, os.path.join(save_directory, shard_file), metadata=metadata)
             else:
+                shard.update(metadata)
                 save_function(shard, os.path.join(save_directory, shard_file))
```

Comment:
For safetensor, we can save the metadata seperately but for torch saving, I had to add the metadata to the shard. Maybe there is a better way to deal with that.